### ISSUE TITLE
WGAN: what does the critic loss represent?

### ISSUE THREAD

**Issue by lennert130994** [eriklindernoren/Keras-GAN#180]
According to the WGAN paper the Wasserstein estimate is equal to:
![image](https://user-images.githubusercontent.com/23079356/63351261-33cea180-c35f-11e9-8e62-06b353ab22cc.png)

In this WGAN implementation: d_loss = 0.5 * np.add(d_loss_fake, d_loss_real)
Then finally the print function outputs D_loss = 1-d_loss

Can someone explain to me how d_loss in this implementation relates to the Wasserstein estimate from the original paper?

**Comment by lennert130994** [comment]
Thanks for your reply @evertonaleixo.
The way i see it the objective function of the critic is to maximize the 'distance' between the training data and the generated data, while the generator is trying to do the opposite. Based on this, I would expect the loss to start at a high possitive value and converge towards zero as the generator starts generating better samples. What I do not understand is why we need to plot 1-d_loss to get the Wasserstein estimate and not simply d_loss (as its trying to maximize that). Perhaps i do not understand your answer fully, so maybe you can elaborate?

**Comment by Lorenzo-Mazza** [comment]
That's my question as well. Is there a reason? I have not found any reference to 1-D Loss anywhere else
